src/OpenTelemetry.Instrumentation.ConfluentKafka/InstrumentedConsumer.cs

```diff
@@ -308,70 +293,59 @@ private static void GetTags(string topic, out TagList tags, int? partition = nul
         }
     }
 
+    private static void UpdateActivityWithConsumeResults(Activity? activity, ConsumeResult<TKey, TValue>? consumeResult)
+    {
+        if (activity == null || consumeResult == null)
+        {
+            return;
+        }
+
+        if (!consumeResult.IsPartitionEOF)
+        {
+            activity.SetTag(SemanticConventions.AttributeMessagingKafkaDestinationPartition, consumeResult.TopicPartition.Partition.Value);
+            activity.SetTag(SemanticConventions.AttributeMessagingDestinationName, consumeResult.Topic);
+            activity.SetTag(SemanticConventions.AttributeMessagingKafkaMessageOffset, consumeResult.Offset.Value);
+        }
+    }
+
     private static void RecordReceive(TopicPartition topicPartition, TimeSpan duration, string? errorType = null)
     {
+        if (!ConfluentKafkaCommon.ReceiveMessagesCounter.Enabled &&
+            !ConfluentKafkaCommon.ReceiveDurationHistogram.Enabled)
+        {
+            return;
+        }
+
         GetTags(topicPartition.Topic, out var tags, partition: topicPartition.Partition, errorType);
 
         ConfluentKafkaCommon.ReceiveMessagesCounter.Add(1, in tags);
```

Comment:
is it always 1? If so, there is there a point in reporting it - it's derived from duration

> This metric is [required](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/metrics/metric-requirement-level.md#required) for batch receive operations. It's [opt-in](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.21.0/specification/metrics/metric-requirement-level.md#opt-in) when the messaging system does not support batch receive since the message count can be derived from the messaging.receive.duration histogram.

https://github.com/open-telemetry/semantic-conventions/blob/v1.24.0/docs/messaging/messaging-metrics.md#metric-messagingreceivemessages